src/Workspaces/CoreTest/SolutionTests/SolutionTests.cs

```diff
@@ -3282,7 +3282,7 @@ public async Task TestDocumentFileAccessFailureMissingFile()
             Assert.Equal("", text.ToString());
 
             // Verify invariant: The compilation is guaranteed to have a syntax tree for each document of the project (even if the contnet fails to load).
-            var compilation = await solution.State.GetCompilationAsync(doc.Project.State, CancellationToken.None).ConfigureAwait(false);
+            var compilation = await solution.CompilationState.GetCompilationAsync(doc.Project.State, CancellationToken.None).ConfigureAwait(false);
```

Comment:
```suggestion
            var compilation = await project.GetCompilationAsync(CancellationToken.None).ConfigureAwait(false);
```

You preserved behavior, but it's tremendously suspicious this test is reaching into the project, grabbing the state, and then asking for the compilation matching that state.

Reply:
(I'm actually confused why that GetCompilationAsync method even exists that takes a state -- if you look at the old implementation it's ignoring the state other than looking at the ID)

Reply:
Yeah, it is weird.  Though it uses the ProjectState for two bits of info.  The ID *and* "SupportsCompilation".  

I will fixup the test though.